internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -401,7 +417,11 @@ func (t *Translator) translateBackendTrafficPolicyForRoute(
 	}
 
 	// Apply IR to all relevant routes
-	for _, x := range xdsIR {
+	for key, x := range xdsIR {
+		if gatewayNN != nil && key != t.IRKey(*gatewayNN) {
```

Comment:
can you help explain the need for this ?

this func eventually calls
```
	prefix := irRoutePrefix(route)
	for _, tcp := range x.TCP {
		for _, r := range tcp.Routes {
			if strings.HasPrefix(r.Destination.Name, prefix) {
			```

Reply:
which does some filtering, can we reuse/improve it ?

Reply:
if you take a look at the test case:
- httproute had two parents
- only 1 gateway has parent BTP
- with prev logic, if `gateway2` process after `gateway1`, it will override the merge result.

Reply:
thanks for explaining this, can you add a comment to improve readibility